crates/rpc/rpc-types/src/eth/trace/tracerequest.rs

```diff
@@ -0,0 +1,62 @@
+//! Builder style functions for `trace_call`
+
+use crate::{state::StateOverride, trace::parity::TraceType, BlockOverrides, CallRequest};
+use reth_primitives::BlockId;
+use std::collections::HashSet;
+
+use serde::{Deserialize, Serialize};
+
+/// Trace Request builder style function implementation
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TraceRequest {
+    /// call request object
+    pub call: CallRequest,
+    /// trace types
+    pub trace_types: HashSet<TraceType>,
+    block_id: Option<BlockId>,
+    state_overrides: Option<StateOverride>,
+    block_overrides: Option<Box<BlockOverrides>>,
```

Comment:
we eventually need to consume all of those fields, so they should all be public, this way we can get rid of the clones